core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala

```diff
@@ -288,10 +288,10 @@ private[ui] class StagePage(parent: StagesTab, store: AppStatusStore) extends We
 
         val executorOverhead = serializationTime + deserializationTime
         val executorRunTime = if (taskInfo.duration.isDefined) {
-          totalExecutionTime - executorOverhead - gettingResultTime
+          math.max(totalExecutionTime - executorOverhead - gettingResultTime - schedulerDelay, 0)
```

Comment:
I am not familiar with the Core module.
But is it possible that `totalExecutionTime - executorOverhead - gettingResultTime - schedulerDelay` is negative?
If so, is the calculation correct?

Reply:
It is impossible.
I write math.max(xxx, 0) here,  just be consistent with the calculation of the executorComputingTimeProportion 
L297:
 val executorComputingTimeProportion =
          math.max(100 - schedulerDelayProportion - shuffleReadTimeProportion -
            shuffleWriteTimeProportion - serializationTimeProportion -
            deserializationTimeProportion - gettingResultTimeProportion, 0)

Reply:
In fact,

executorComputingTime = executorRunTime - shuffleReadTime - shuffleWriteTime=totalExecutionTime - executorOverhead - gettingResultTime- schedulerDelay-shuffleReadTime - shuffleWriteTime

executorComputingTimeProportion = (totalExecutionTime - executorOverhead - gettingResultTime - schedulerDelay-shuffleReadTime - shuffleWriteTime) / totalExecutionTime * 100

but in the old code executorComputingTime does not minus schedulerDelay

Reply:
I think it's fine for consistency and perhaps to account for rounding errors or something. I'm OK with it as the current calculation does seem obviously wrong in at least one case.